### PR TITLE
Add support for validated PUT requests

### PR DESCRIPTION
**Pull request by smn**
This builds on PR #4

### CODE REVIEW

**Comment by smn** [comment]
This requires https://github.com/universalcore/elastic-git/pull/54 to land.

**Comment by smn** [comment]
@Rizziepit & @miltontony one to review on monday

**Comment by smn** [comment]
Ah my validator needs a test

**Comment by smn** [comment]
:wave: :eyes:

**Comment by Rizziepit** [comment]
fwiw, I've gone over this and it looks fine

**Comment by smn** [comment]
I'm a bit uncertain about what an appropriate response is to the HTTP `PUT` and `DELETE` verbs. `POST`s should return in a redirect since it's the creation of a new object which can be found in a new location. `PUT` updates an object at the given location. So from an API user's perspective a redirect may be expected but a redirect to a `GET` makes not much since since it's the same as what's been written (which is why I'm returning the body). The other question for the `DELETE` is that I'm unsure what we're supposed to be redirecting to in that case, again I'm returning the object that was just deleted.

You all have any thoughts on this?

**Comment by Rizziepit** [comment]
I've never worked with a RESTful API that redirects, whether on `DELETE` or `POST`. So I'm not sure if that is expected? Usually the body isn't returned, rather the cleaned/transformed object representation that has been saved/deleted/updated. It _could_ be the same as the body. In the case of `POST`, the returned object representation would contain a "detail" URI that you can choose to `GET`. In the case of `DELETE` that URI would be missing.

On a side note, django-tastypie lets you configure whether you want `PUT` and `POST` to return the object representation - default is `True`.

**Comment by Rizziepit** [comment]
Are you using response codes like 201 and 204?

**Comment by miltontony** [comment]
:+1:  As for the responses, [RFC-7231](http://tools.ietf.org/html/rfc7231#section-4.3) has some recommendations on what the response codes can be.
TL;DR
Only `20x` codes are to be expected. With or without a body is up to you.

**Comment by smn** [comment]
@Rizziepit good point, not setting response codes anywhere atm.

**Comment by smn** [comment]
I'm leaving it as `200` since both `PUT` and `DELETE` return a body and `200` means the request was accepted immediately. Once I ad a `POST` it'll return with a `201` for newly created content.

**Comment by smn** [comment]
@Rizziepit & @miltontony ready for re-review

**Comment by Rizziepit** [comment]
+1
